plaid/src/apis/github/code.rs

```diff
@@ -0,0 +1,61 @@
+use std::collections::HashMap;
+
+use super::Github;
+use crate::apis::{github::GitHubError, ApiError};
+
+impl Github {
+    /// Search for files with given name across a GH organization and return the result.
+    /// See https://docs.github.com/en/rest/search/search?apiVersion=2022-11-28#search-code for more details
+    pub async fn search_file_in_org_code(
+        &self,
+        params: &str,
+        module: &str,
+    ) -> Result<String, ApiError> {
+        let request: HashMap<&str, &str> =
+            serde_json::from_str(params).map_err(|_| ApiError::BadRequest)?;
+
+        let org = self.validate_org(request.get("org").ok_or(ApiError::BadRequest)?)?;
+        let filename =
+            self.validate_filename(request.get("filename").ok_or(ApiError::BadRequest)?)?;
+        let per_page: u8 = request
+            .get("per_page")
+            .unwrap_or(&"30")
+            .parse::<u8>()
+            .map_err(|_| ApiError::BadRequest)?;
+        let page: u16 = request
+            .get("page")
+            .unwrap_or(&"1")
+            .parse::<u16>()
+            .map_err(|_| ApiError::BadRequest)?;
+
+        if per_page > 100 {
+            // GitHub supports up to 100 results per page
+            return Err(ApiError::BadRequest);
+        }
+
+        info!(
+            "Finding all files called [{filename}] in organization [{org}] on behalf of [{module}]"
+        );
+
+        // Build the search query
+        let query = urlencoding::encode(&format!("{} in:path org:{}", filename, org)).to_string();
```

Comment:
I'm thinking we should provide a generic search API. Any thoughts on the pros and cons?

Reply:
Yes, that's a fair point. I was considering this but then I stumbled on the rate limitation for the search code endpoint (see [here](https://docs.github.com/en/rest/search/search?apiVersion=2022-11-28#rate-limit)).
We get only 10 requests per minute, so I increased the page size to its max, in an effort to make as few calls as we can.

Basically, my fear is that if we allow very generic searches, rules might start experiencing failures and everything kind of grinds to a halt when it comes to GH searches. If we narrow down the scope, we have more of a chance. 

What do you think?

Reply:
@obelisk I added support for searching only in a repository, instead of searching in the entire org and _then_ filtering. So we make less calls and we should stay farther away from rate limitations.